### PR TITLE
fix(ingestion/airflow-plugin): fixed missing inlet/outlets

### PR DESCRIPTION
**Pull request by dushayntAW**
## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: metadata-ingestion-modules/airflow-plugin/src/datahub_airflow_plugin/_datahub_ol_adapter.py [hunk 1]
```diff
@@ -14,10 +14,15 @@
...
 
 def translate_ol_to_datahub_urn(ol_uri: OpenLineageDataset) -> str:
-    namespace = ol_uri.namespace
-    name = ol_uri.name
-
-    scheme, *rest = namespace.split("://", maxsplit=1)
-
-    platform = OL_SCHEME_TWEAKS.get(scheme, scheme)
-    return builder.make_dataset_urn(platform=platform, name=name)
+    if not hasattr(ol_uri, "namespace") or not ol_uri.namespace:
```
Comment (hsheth2) [inline metadata-ingestion-modules/airflow-plugin/src/datahub_airflow_plugin/_datahub_ol_adapter.py:17]: when does this happen? what should the type annotation of this function be changed to?

File: metadata-ingestion-modules/airflow-plugin/tests/integration/dags/custom_operator_dag.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+with DAG(
+    "custom_operator_dag",
+    default_args=default_args,
+    description="An example dag with custom operator",
+    schedule_interval=None,
+    tags=["example_tag"],
+    catchup=False,
+    default_view="tree",
+) as dag:
+    custom_task = CustomOperator(
```
Comment (hsheth2) [inline metadata-ingestion-modules/airflow-plugin/tests/integration/dags/custom_operator_dag.py:25]: please just define and use the custom operator in the same file - no need to import it from elsewhere

File: metadata-ingestion-modules/airflow-plugin/tests/integration/plugins/custom_operator.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+        input_tables = ["mydb.schema.tableA", "mydb.schema.tableB"]
+        output_tables = ["mydb.schema.tableD"]
+
+        inlets, outlets = self._get_sf_lineage(input_tables, output_tables)
+
+        context["ti"].task.inlets = inlets
+        context["ti"].task.outlets = outlets
+
+    @staticmethod
+    def _get_sf_lineage(input_tables, output_tables):
```
Comment (hsheth2) [inline metadata-ingestion-modules/airflow-plugin/tests/integration/plugins/custom_operator.py:31]: add type annotations

File: metadata-ingestion-modules/airflow-plugin/tests/integration/test_plugin.py [hunk 4]
```diff
@@ -132,6 +136,7 @@ def _run_airflow(
         # Point airflow to the DAGs folder.
         "AIRFLOW__CORE__LOAD_EXAMPLES": "False",
         "AIRFLOW__CORE__DAGS_FOLDER": str(dags_folder),
+        "AIRFLOW__CORE__PLUGINS_FOLDER": str(plugins_folder),
```
Comment (hsheth2) [inline metadata-ingestion-modules/airflow-plugin/tests/integration/test_plugin.py:139]: once you change the code to define the CustomOperator in the same file as the dag, this won't be necessary

**Review by hsheth2 — CHANGES_REQUESTED** [review]

File: metadata-ingestion-modules/airflow-plugin/src/datahub_airflow_plugin/_datahub_ol_adapter.py (hunk 1 above)
Comment (dushayntAW) [reply metadata-ingestion-modules/airflow-plugin/src/datahub_airflow_plugin/_datahub_ol_adapter.py]: this is no longer required and reverted back to the original

File: metadata-ingestion-modules/airflow-plugin/tests/integration/test_plugin.py [hunk 5]
```diff
@@ -31,6 +31,7 @@
 IS_LOCAL = os.environ.get("CI", "false") == "false"
 
 DAGS_FOLDER = pathlib.Path(__file__).parent / "dags"
+PLUGINS_FOLDER = pathlib.Path(__file__).parent / "plugins"
```
Comment (hsheth2) [inline metadata-ingestion-modules/airflow-plugin/tests/integration/test_plugin.py:34]: remove this line

Comment (hsheth2) [reply metadata-ingestion-modules/airflow-plugin/tests/integration/test_plugin.py:34]:
```suggestion
```

**Review by hsheth2 — APPROVED** [review]

**Pull request merged**